packages/core-utils/src/itinerary.ts

```diff
@@ -329,30 +340,41 @@ export function getElevationProfile(
   let traversed = 0;
   let gain = 0;
   let loss = 0;
-  let previous = null;
+  let previous: ElevationProfileComponent | null = null;
   const points = [];
   steps.forEach(step => {
-    if (!step.elevation || step.elevation.length === 0) {
+    // Support for old REST response data (in step.elevation)
+    const stepElevationProfile =
+      step.elevationProfile ||
+      (Array.isArray(step.elevation) &&
+        step.elevation?.map<ElevationProfileComponent>(
+          mapOldElevationComponentToNew
+        ));
+
+    if (!stepElevationProfile || stepElevationProfile.length === 0) {
       traversed += step.distance;
       return;
     }
-    for (let i = 0; i < step.elevation.length; i++) {
-      const elev = step.elevation[i];
+    for (let i = 0; i < stepElevationProfile.length; i++) {
+      const elev = stepElevationProfile[i];
       if (previous) {
-        const diff = (elev.second - previous.second) * unitConversion;
+        const diff = (elev.elevation - previous.elevation) * unitConversion;
         if (diff > 0) gain += diff;
         else loss += diff;
       }
-      if (i === 0 && elev.first !== 0) {
+      if (i === 0 && elev.distance !== 0) {
         // console.warn(`No elevation data available for step ${stepIndex}-${i} at beginning of segment`, elev)
```

Comment:
Is there a point to keeping this commented out `console.warn`?

Reply:
IDK but it's not really relevant to this PR so I don't wanna touch it